File: day6/golang/quote/pkg/models/dto/eventdto.go
package dto

import (
	"time"
)

type Event struct {
	Id          int       `json:"id"`
	Day         int       `json:"day" validate:"required,gte=1,lte=31"`
	Month       int       `json:"month"`
	Year        int       `json:"year"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

File: day6/java/quote/src/main/java/com/ns/quote/models/dto/EventDto.java
package com.ns.quote.models.dto;

import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.time.ZonedDateTime;

@Getter
@Setter
public class EventDto {
    private int id;
    @NotNull
    @Min(1)
    @Max(31)
    private int day;
    private int month;
    private int year;
    private String title;
    private String description;
    private ZonedDateTime createdAt;
    private ZonedDateTime updatedAt;
}
